odoo/modules/registry.py

```diff
@@ -276,10 +276,10 @@ def setup_models(self, cr):
         def transitive_dependencies(field, seen=[]):
             if field in seen:
                 return
-            for seq1 in dependencies[field]:
-                yield seq1
-                exceptions = (Exception,) if field.base_field.manual else ()
-                with ignore(*exceptions):
+            exceptions = (Exception,) if field.base_field.manual else ()
+            with ignore(*exceptions):
+                for seq1 in dependencies[field]:
+                    yield seq1
```

Comment:
I see, you are generalizing the use of `ignore()`.  But the latter is already more complex than necessary: the only exception being caught is a trivial key error in `dependencies`.  Just replace the key access `[field]` by a `.get(field)` and you get something that never crashes:
```python
def transitive_dependencies(field, seen=[]):
    if field in seen:
        return
    for seq1 in dependencies.get(field, ()):
        yield seq1
        for seq2 in transitive_dependencies(seq1[-1], seen + [field]):
            yield concat(seq1[:-1], seq2)
```

Reply:
@rco-odoo Done  👍